### PR TITLE
audit: enforce https on [*.]archive.org

### PR DESCRIPTION
**Pull request by vszakats**

### CODE REVIEW

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -439,7 +439,8 @@ def audit_homepage
          %r{^http://[^/.]+\.tools\.ietf\.org},
          %r{^http://www\.gnu\.org/},
          %r{^http://code\.google\.com/},
-         %r{^http://bitbucket\.org/}
+         %r{^http://bitbucket\.org/},
+         %r{^http://([^/]*\.|)archive\.org}
```
Comment (bfontaine) [inline Library/Homebrew/cmd/audit.rb]: We don’t need to capture the group, so you can use `(?:[^/]*\.)?`.

Comment (vszakats) [inline Library/Homebrew/cmd/audit.rb]: Nice, thanks!

Comment (vszakats) [inline Library/Homebrew/cmd/audit.rb]: Updated for `bintray.com` as well. (where I copied this regexp from)

**Comment by mikemcquaid** [comment]
Thanks @vszakats!

**Pull request closed without merging**